### PR TITLE
[onduty] Add Onduty Bot to handle rotation updates and update teams

### PR DESCRIPTION
**Pull request by rcebulko**
Part of #778

### CODE REVIEW

File: onduty/src/bot.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const teamName = this.rotationTeams[type];
+    if (!teamName) {
+      this.logger.error(`[updateRotation] Missing ${type} rotation team`);
+      return;
+    }
+    this.logger.info('[updateRotation] Updating ${type} rotation');
+
+    const members = await this.github.getTeamMembers(teamName);
+    const currentRotation = Object.values(rotation)
+      .filter(Boolean)
```
Comment (estherkim) [inline onduty/src/bot.ts:42]: Curious, why is `filter(Boolean)` needed?

File: onduty/types/index.d.ts [hunk 2]
```diff
@@ -22,4 +22,13 @@ declare module 'onduty' {
     error(message: string, ...extraInfo: unknown[]): void;
     info(message: string, ...extraInfo: unknown[]): void;
   }
+
+  export interface Rotation {
+    primary: string;
+    secondary: null | string;
+  }
```
Comment (estherkim) [inline onduty/types/index.d.ts:29]: GitHub teams don't have a concept for which members are primary or secondary, right? Could this be an array of usernames instead?

Comment (rcebulko) [reply onduty/types/index.d.ts:29]: This is the format of the payload from the rotation reporter tool :)

File: onduty/src/bot.ts (hunk 1 above)
Comment (rcebulko) [reply onduty/src/bot.ts:42]: `secondary` may not always be present in the payload from the rotation reporting tool

**Pull request merged**